### PR TITLE
Jon/task/swift style guide update

### PR DESCRIPTION
**Pull request by jonthejon**
As discussed in our last team meeting and also over chat on this [thread](https://ecobeeteam.slack.com/archives/G2D5YDREU/p1573834198020400), I'm suggesting here a change on how we write our single return guard statements.

Still open for discussion and suggestions though! :) So feel free to say what's on your mind about this.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -931,17 +924,27 @@ guard let thingThree = thingThree else {
...
 // PREFERRED
-guard let thingOne = thingOne else {
-    return
+guard let thingOne = thingOne else { return nil }
+
+guard let name = json["name"] as? String,
+  let coordinatesJSON = json["coordinates"] as? [String: Double],
+  let latitude = coordinatesJSON["lat"],
+  let longitude = coordinatesJSON["lng"],
+  let mealsJSON = json["meals"] as? [String] else { return }
```
Comment (miwandn) [inline README.md]: Hmm I am ok with one liner guards but when you have multiple properties, I prefer the return to be on the second line i.e. 
```
guard let name = json["name"] as? String,
  let coordinatesJSON = json["coordinates"] as? [String: Double],
  let latitude = coordinatesJSON["lat"],
  let longitude = coordinatesJSON["lng"],
  let mealsJSON = json["meals"] as? [String] else
 {
    return 
 }

File: README.md [hunk 2]
```diff
@@ -837,9 +837,8 @@ func eatDoughnut(at index: Int) {
 
 ```swift
 // PREFERRED
-guard let monkeyIsland = monkeyIsland else {
-    return
-}
+guard let monkeyIsland = monkeyIsland else { return }
```
Comment (miwandn) [inline README.md:840]: Would this also apply if you were returning a value other then void?

File: README.md (hunk 1 above)
Comment (jonthejon) [reply README.md]: Ya, that's valid point. Not my preference but I see where you're coming from. I'll take that back to the team so we can come with something that's agreed by the majority.

File: README.md (hunk 2 above)
Comment (jonthejon) [reply README.md:840]: In this current proposal yes. See line 932 with a `// PREFERRED` example:
`guard let thingOne = thingOne else { return nil }`

**Review by joedrew — APPROVED** [review]
My opinion doesn't count here :)

File: README.md (hunk 1 above)
Comment (cswelin) [reply README.md]: 100% with Miwand, any multi show following with a new line

File: README.md (hunk 2 above)
Comment (jonthejon) [reply README.md:840]: Ya, as we talked before, your comment got me thinking. In the sense that `return nil` is very different than `return foo?.value != nil ? true : false`.

Updated the proposal for only the first case (reduced return values)

Change on [52319ab](https://github.com/ecobee/swift-style-guide/pull/3/commits/52319ab2b7af77e6ce19041343285762652a735e).

File: README.md (hunk 1 above)
Comment (jonthejon) [reply README.md]: Looks like this is what the people want. :)
Change on [52319ab](https://github.com/ecobee/swift-style-guide/pull/3/commits/52319ab2b7af77e6ce19041343285762652a735e).

**Review by miwandn — APPROVED** [review]

**Review by vpolsinelli — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -913,9 +908,7 @@ if let woodchuck = woodchuck, canChuckWood(woodchuck) {
 // combined because we just return
 guard let thingOne = thingOne,
     let thingTwo = thingTwo,
-    let thingThree = thingThree else {
-    return
-}
+    let thingThree = thingThree else { return }
```
Comment (chengmingecobee) [inline README.md]: I feel like this should be the same case (multi-line) and therefore we should not return in the same line. What do you think?

**Review by chengmingecobee — COMMENTED** [review]
Rest looks good

File: README.md (hunk 3 above)
Comment (jonthejon) [reply README.md]: Good eye man! Changed it back here [4e3e2c0](https://github.com/ecobee/swift-style-guide/pull/3/commits/4e3e2c0c96da27d7ecc0ce2885e14f1b1a3a2a53)

**Review by chengmingecobee — COMMENTED** [review]
👍

**Review by chengmingecobee — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -931,17 +927,32 @@ guard let thingThree = thingThree else {
...
+guard let thingOne = thingOne else { return nil }
+
+guard let name = json["name"] as? String,
+  let coordinatesJSON = json["coordinates"] as? [String: Double],
+  let latitude = coordinatesJSON["lat"],
+  let longitude = coordinatesJSON["lng"],
+  let mealsJSON = json["meals"] as? [String] else 
+{
+  return 
+}
```
Comment (yootsubo) [inline README.md:944]: Doesn't seem like tabbed correctly

**Review by sonerecobee — APPROVED** [review]

**Review by yootsubo — APPROVED** [review]

**Pull request merged**